src/vos/vos_pool_scrub.c

```diff
@@ -272,20 +272,27 @@ sc_get_rec_in_chunk_at_idx(const struct scrub_ctx *ctx, uint32_t i)
 static void
 sc_wait_until_should_continue(struct scrub_ctx *ctx)
 {
+	if (sc_cont_is_stopping(ctx))
+		return;
+
 	if (sc_mode(ctx) == DAOS_SCRUB_MODE_TIMED) {
 		struct timespec	now;
 		uint64_t	msec_between;
 
 		d_gettime(&now);
 		while ((msec_between = sc_get_ms_between_scrubs(ctx)) > 0) {
 			d_tm_set_gauge(ctx->sc_metrics.scm_next_csum_scrub, msec_between);
+			if (sc_cont_is_stopping(ctx))
```

Comment:
Strictly, checking stopping before d_tm_set_gauge() or after sc_sleep() seems better. Anyway, not serious, can be adjusted latter.

Reply:
You are right. fixed.